src/push/utils.ts

```diff
@@ -90,14 +90,51 @@ export function printBytes(bytes: number) {
   return `${bytes.toFixed(1)} ${BYTE_UNITS[exponent]}`;
 }
 
-export function getChunks<T>(arr: Array<T>, size: number): Array<T[]> {
+export function getChunks(arr: string[], size: number): string[][] {
   const chunks = [];
   for (let i = 0; i < arr.length; i += size) {
     chunks.push(arr.slice(i, i + size));
   }
   return chunks;
 }
 
+export function getSizedChunks<T extends { id?: string }>(
+  arr: Array<T>,
+  sizes: Map<string, number>,
+  maxChunkSizeKB: number,
+  maxChunkItems: number
+): Array<T[]> {
+  const chunks: Array<T[]> = [];
+  let currentChunk: T[] = [];
+  let currentSize = 0;
+
+  for (const item of arr) {
+    const sizeKb = item.id ? sizes.get(item.id) / 1024 : 1;
```

Comment:
This is size in bytes, Lets round it off, so can be Math.round(sizes.get(item.id) / 1000)